File: Medium/2379. Minimum Recolors to Get K Consecutive Black Blocks/minimumRecolors.cpp
class Solution {
    public:
        int minimumRecolors(string blocks, int k) {
            uint8_t firstPointer = 0, lastPointer = k;
            int sumWhites = 0;
            for(; firstPointer < k; firstPointer++){
                if (blocks[firstPointer] == 'W'){
                    sumWhites++;
                }
            }
            int minWhites = sumWhites;
            for(firstPointer = 0; lastPointer < blocks.size(); firstPointer++, lastPointer++){
                if (blocks[firstPointer] == 'W'){
                    sumWhites--;
                }
                if (blocks[lastPointer] == 'W'){
                    sumWhites++;
                }
                minWhites = min(minWhites, sumWhites);
            }
            return minWhites;
        }
    };

File: Medium/2379. Minimum Recolors to Get K Consecutive Black Blocks/minimumRecolors.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        sum_whites = 0
        for i in range(k):
            if blocks[i] == "W":
                sum_whites += 1

        min_whites = sum_whites

        for i in range(1, len(blocks) - k + 1):
            if blocks[i - 1] == "W":
                sum_whites -= 1
            if blocks[i + k - 1] == "W":
                sum_whites += 1
            min_whites = min(min_whites, sum_whites)

        return min_whites
